engine/tests/phpunit/Elgg/Filesystem/AdapterServiceTest.php

```diff
@@ -0,0 +1,188 @@
+<?php
+namespace Elgg\Filesystem;
+use Elgg\Filesystem\Adapter\Adapter;
+use Elgg\Filesystem\AdapterService;
+use Gaufrette\Filesystem;
+use Gaufrette\File;
+use Gaufrette\Adapter\InMemory as GaufretteMemory;
+
+class AdapterServiceTest extends \PHPUnit_Framework_TestCase {
+	public function singleAdapter() {
+		$l = $this->getMock('Elgg\Logger', [], [], '', false);
+		$service = new AdapterService($l);
+
+		$adapter = new inMemoryAdapter([]);
+		$name = "in_memory";
+		return [[$service, $adapter, $name]];
+	}
+	
+	public function multipleAdapters() {
+		$adapters = [];
+		$l = $this->getMock('Elgg\Logger', [], [], '', false);
+		$service = new AdapterService( $l);
+		
+		for ($i=1; $i<=3; $i++) {
+			$adapter = new inMemoryAdapter([]);
+			$name = "in_memory_$i";
+			$adapters[$name] = $adapter;
+		}
+			
+		return [[$service, $adapters]];
+	}
+	
+	/**
+	 * @dataProvider singleAdapter
+	 */
+	function testCanSetAdapter($s, $adp, $name) {
+		$this->assertTrue($s->set($name, $adp));
+	}
+	
+	/**
+	 * @dataProvider singleAdapter
+	 */
+	function testSetThrowsOnSameName($s, $adp, $name) {
+		$this->assertTrue($s->set($name, $adp));
+		$this->setExpectedException('\UnexpectedValueException');
+		$s->set($name, $adp);
+	}
+	
+	/**
+	 * @dataProvider multipleAdapters
+	 */
+	function testGet($s, $adapters) {
```

Comment:
Can you be explicit about what behavior you're testing for? Even if it feels like you're just returning the PHPdoc for the get method?